File: app/src/main/java/com/ed/redditapp/lib/api/standard/StandardComment.java
package com.ed.redditapp.lib.api.standard;

import com.ed.redditapp.lib.api.Comment;

import org.json.JSONException;

import lombok.Getter;

@Getter
public class StandardComment implements Comment {
    private final String username;
    private final String body;
    private final long timestamp;
    private final int points;
    private final int indent;

    public StandardComment(StandardJsonComment jsonComment) throws JSONException {
        username = jsonComment.getData().getString("author");
        body = jsonComment.getData().getString("body_html");
        timestamp = jsonComment.getData().getLong("created_utc");
        points = jsonComment.getData().getInt("ups");
        indent = jsonComment.getIndent();
    }
}

File: app/src/main/java/com/ed/redditapp/lib/api/standard/StandardComment.kt
package com.ed.redditapp.lib.api.standard

import com.ed.redditapp.lib.api.Comment

class StandardComment(jsonComment: StandardJsonComment) : Comment {
    override val username = jsonComment.data.getString("author")
    override val body = jsonComment.data.getString("body_html")
    override val timestamp = jsonComment.data.getLong("created_utc")
    override val points = jsonComment.data.getInt("ups")
    override val indent = jsonComment.indent
}